src/base/static/components/templates/map.tsx

```diff
@@ -12,6 +14,7 @@ import LeftSidebar from "../organisms/left-sidebar";
 import RightSidebar from "../organisms/right-sidebar";
 import GeocodeAddressBar from "../organisms/geocode-address-bar";
 
+import { IMapViewport } from "../app";
```

Comment:
This is creating a circular dependency, which we should avoid: https://medium.com/content-uneditable/circular-dependencies-in-javascript-a-k-a-coding-is-not-a-rock-paper-scissors-game-9c2a9eccd4bc

I'm not sure where to put `IMapViewport`, but we can consider making a `map.d.ts` file alongside our map duck, and then import it from there.

Reply:
Would it make more sense to define it in an app.d.ts file alongside the `App` component? The map duck doesn't store the viewport anymore, so it might be confusing.

Reply:
The "d.ts" file is used to provide typescript type information about an API that's written in JavaScript, so `app.d.ts` wouldn't make sense.

It seems like the most proper solution here would be to put our MapViewport in a Context. Although we aren't use the Context API yet, we can consider making a new `static/contexts/` dir, and putting `map.tsx` file under that.

Reply:
The "d.ts" file is used to provide typescript type information about an API that's written in JavaScript, so `app.d.ts` wouldn't make sense.

It seems like the most proper solution here would be to put our MapViewport in a Context. Although we aren't using the Context API yet, we can consider making a new `static/contexts/` dir, and putting `map.tsx` file under that.

Reply:
Can we create something like an `interfaces.ts` file and put it in `src/base/static` instead? 

My understanding of the Context API is that it's for sharing state among numerous components at different levels of the hierarchy. Since the map viewport is only used by a couple of components it seems like it might be overkill here.

Reply:
My concern is that `interfaces.ts` would become a dumping grounds, and it's not clear what kinds of interfaces will be stored there. As we port over to TS, we're going to defining lots of interfaces.

I'm not too familiar with Context API, but it seems like a reasonable use case since we're drilling our MapViewport from App > MapTemplate > MainMap, and perhaps other components as well, like the Drawing component, Story Sidebar, etc.

But feel free to use your judgement :+1:

Reply:
I guess I'm not sure what the best thing to do is. Maybe to keep things simple in this PR I'll just define a separate interface in `map.tsx`, which actually might be appropriate because technically the `initialMapViewport` and `mapViewport` differ slightly.

Reply:
And, I guess since we're not sharing interfaces I'll do away with the `I` syntax too :).